usr.bin/top/top.c

```diff
@@ -907,6 +908,20 @@ main(int argc, const char *argv[])
 				}
 				break;
 
+			    case CMD_grep: /* grep command name */
+				new_message(MT_standout,
+				    "Grep command name: ");
+				if (readline(tempbuf1, sizeof(tempbuf1), false) > 0)
+				{
```

Comment:
This { needs to be at the end of the prior line.

Also, if I do /getty<CR> I see just the getty routines. However, if I do /<CR> next, it doesn't reset the display to all processes because readline is return  -1 in this case...

Reply:
And there's a resource leak, both of these can be fixed by free'ing ps.command and setting it to  NULL before the readline...

Reply:
Resetting the display to all processes is handled by /+  That's the behavior in OpenBSD.  I like it because it allows you to back out of a search.

The missing free is fixed.  A const qualifier was added to the process_select struct in fc36f5a7cb968 though I'm unsure why.

Reply:
The const can be removed, I think.